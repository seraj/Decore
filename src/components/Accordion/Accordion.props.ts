import ThemeProps from "../../theme/Theme.props";

export default interface AccordionProps
  extends ThemeProps,
    React.ImgHTMLAttributes<HTMLImageElement> {
  question?: string;
}
