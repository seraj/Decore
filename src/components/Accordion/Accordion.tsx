import * as React from "react";
import StyledAccordion from "./styled/StyledAccordion";
import AccordionProps from "./Accordion.props";

interface AccordionState {
  active: boolean;
}

class Accordion extends React.Component<AccordionProps, AccordionState> {
  public state = {
    active: false
  };
  public toggle = (event: React.MouseEvent) => {
    this.setState({
      active: !this.state.active
    });
  };
  public render() {
    const {
      props: { question, children },
      state: { active },
      toggle
    } = this;
    const classes = active ? "active" : "inactive";
    return (
      <StyledAccordion className={classes}>
        <span className="accordion-label" onClick={toggle}>
          {question}
        </span>
        <span className="accordion-content">{children}</span>
      </StyledAccordion>
    );
  }
}

export default Accordion;
