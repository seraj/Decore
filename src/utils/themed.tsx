import * as React from "react";
import * as renderer from "react-test-renderer";
import Decore from "../components/Decore";
import theme from "../theme";

const themed = (component: JSX.Element) =>
  renderer.create(<Decore theme={theme}>{component}</Decore>);

export default themed;
