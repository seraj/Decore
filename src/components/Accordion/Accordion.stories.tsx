import { withKnobs } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as React from "react";
import wInfo from "../../utils/wInfo";
import Accordion from "./Accordion";

const stories = storiesOf("Components/Accordion", module);
stories.addDecorator(withKnobs);

stories.add(
  "default",
  wInfo()(() => (
    <>
      <Accordion question="Hey!">This is Test</Accordion>
      <Accordion question="Hey!">This is Test</Accordion>
    </>
  ))
);
