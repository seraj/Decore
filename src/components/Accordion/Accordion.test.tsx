import * as React from "react";

import themed from "../../utils/themed";
import Accordion from "./Accordion";

describe("Accordion", () => {
  it("should render correctly", () => {
    const component = themed(<Accordion>Test</Accordion>).toJSON();

    expect(component).toMatchSnapshot();
  });
});
