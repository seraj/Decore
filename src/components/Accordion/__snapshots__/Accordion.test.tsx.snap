// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should render correctly 1`] = `
.c0 {
  overflow: hidden;
  border-bottom: 1px solid #F5F5F5;
}

.c0 .accordion-content {
  display: block;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
  line-height: 20px;
  padding: 0 1em;
}

.c0 .accordion-label {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 1em;
  cursor: pointer;
}

.c0 .accordion-label::after {
  content: "\\276F";
  -webkit-transition: all .35s;
  transition: all .35s;
}

.c0.active .accordion-content {
  -webkit-transform: perspective(350px);
  -ms-transform: perspective(350px);
  transform: perspective(350px);
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
  height: 100%;
  line-height: 30px;
}

.c0.active .accordion-label::after {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c0.inactive .accordion-content {
  -webkit-transform-origin: 50% 0%;
  -ms-transform-origin: 50% 0%;
  transform-origin: 50% 0%;
  -webkit-transform: perspective(250px);
  -ms-transform: perspective(250px);
  transform: perspective(250px);
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
  height: 0;
}

<div
  className="inactive c0"
>
  <span
    className="accordion-label"
    onClick={[Function]}
  />
  <span
    className="accordion-content"
  >
    Test
  </span>
</div>
`;
